rtrlib/rtr/packets.c

```diff
@@ -481,6 +507,27 @@ static bool rtr_pdu_check_size(const struct pdu_header *pdu)
 		if (sizeof(struct pdu_reset_query) == pdu->len)
 			retval = true;
 		break;
+	case ASPA:
+		aspa_pdu = (const struct pdu_aspa *)pdu;
+		min_size = sizeof(struct pdu_aspa);
+		if (aspa_pdu->len < min_size) {
+			RTR_DBG1("PDU is too small to contain ASPA PDU!");
+			break;
+		}
+		
+		// Check if the PDU really contains the ASPA PDU
+		uint16_t asn_count = ntohs(aspa_pdu->provider_count);
+		
+		RTR_DBG("provider_asn_count: %u", asn_count);
+		// ASN is 4 bytes each
+		min_size += asn_count * 4;
```

Comment:
magic number just replace the 4 by `sizeof(uint32_t)`

Reply:
wouldn't `sizeof(aspa_pdu->provider_asns[0])` be better?

Reply:
or this, but I think my suggestion is still cleaner.